AccountMasterKey.js

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+import cryptoLd from 'crypto-ld';
+const {Ed25519KeyPair} = cryptoLd;
+import {Kek} from './Kek.js';
+import {Hmac} from './Hmac.js';
+
+const VERSIONS = ['recommended', 'fips'];
+
+export class AccountMasterKey {
+  /**
+   * Creates a new instance of an AccountMasterKey. This function should never
+   * be called directly. Use one of these methods to create an AccountMasterKey
+   * instance:
+   *
+   * `AccountMasterKey.fromSecret`
+   * `AccountMasterKey.fromBiometric`
+   * `AccountMasterKey.fromFido`
+   *
+   * @param {String} accountId the ID of the account associated with this
+   *   master key.
+   * @param {Object} signer an API for creating digital signatures using the
+   *   master authentication key.
+   * @param {KmsService} kmsService the kmsService to use to perform key
+   *   operations.
+   * @param {String} kmsPlugin the ID of the KMS plugin to use.
+   *
+   * @return {AccountMasterKey}.
+   */
+  constructor({accountId, signer, kmsService, kmsPlugin}) {
+    this.accountId = accountId;
+    this.signer = signer;
+    this.kmsService = kmsService;
+    this.kmsPlugin = kmsPlugin;
+  }
+
+  /**
+   * Generates a key. The key can be a key encryption key (KEK) or an HMAC
+   * key. It can be generated using a FIPS-compliant algorithm or the latest
+   * recommended algorithm.
+   *
+   * @param {String} type the type of key to create (`hmac` or `kek`).
+   * @param {String} version `fips` to use FIPS-compliant ciphers,
+   *   `recommended` to use the latest recommended ciphers.
+   *
+   * @return {Promise<Object>} resolves to a Kek or Hmac instance.
+   */
+  async generateKey({type, version = 'recommended'}) {
+    _assertVersion(version);
+
+    // for the time being, fips and recommended are the same; there is no
+    // other standardized key wrapping algorithm
+    let Class;
+    if(type === 'hmac') {
+      type = 'HS256';
+      Class = Hmac;
+    } else if(type === 'kek') {
+      type = 'AES-KW';
+      Class = Kek;
+    } else {
+      throw new Error(`Unknown key type "${type}".`);
+    }
+
+    const {kmsService, kmsPlugin, kmsPlugin: plugin, signer} = this;
```

Comment:
why the `plugin` alias if it's just used once below?

Reply:
I guess I'm asking why not just have similar rename code in the call below.

Reply:
Because namespacing. The kms service knows the plugin is a kms one ... the key class could potentially use plugins for some other thing.

Reply:
I'll just resolve this.  Seems this is the style used elsewhere too.